### PR TITLE
"solana program set-upgrade-authority" uses SetAuthorityChecked instruction by default

### PR DESCRIPTION
**Pull request by 0xripleys**
#### Problem
Accompanying CLI change for https://github.com/solana-labs/solana/pull/28424
(Original issue: https://github.com/solana-labs/solana/issues/27932)

#### Summary of Changes
- solana program set-upgrade-authority requires the new authority to sign by default. this behavior can be overwritten by the flag `--no-sign-with-new-authority`

Added a basic unit test and also did some manual testing, note that the feature gate for SetAuthorityChecked doesn't seem to have been enabled yet.
```
# provide new authority as pubkey only, the cli fails and complains about no signer
cargo run program set-upgrade-authority  GxSxDvJb7BEe32r93btUXgU41JEVYRr1YgHwZg94fHSk --new-upgrade-authority Gmwr4u7qyjtzbjwmk4XrWEWyn6aA6rWhAA6BUoFNdikG -u d

Error: Dynamic program error: missing signature for supplied pubkey: Gmwr4u7qyjtzbjwmk4XrWEWyn6aA6rWhAA6BUoFNdikG

# provide new authority as keypair, the transaction lands on devnet (but fails as expected since feature gate is disabled)
# https://solscan.io/tx/iKdbztjZXfQAhHa8eBPZfr5sMArVEdqRAFY14aNAvMogpY2q8Mtijd6BGMEUkyHgnceCCJmo76TXdxu3fdKSdup?cluster=devnet
cargo run program set-upgrade-authority  GxSxDvJb7BEe32r93btUXgU41JEVYRr1YgHwZg94fHSk --new-upgrade-authority test_1.json -u d
Error: Setting authority failed: Error processing Instruction 0: invalid instruction data
```

### CODE REVIEW

File: cli/src/program.rs [hunk 1]
```diff
@@ -1171,6 +1215,51 @@ fn process_set_authority(
...
+    tx.try_sign(
+        &[config.signers[0], authority_signer, new_authority_signer],
+        blockhash,
+    )?;
+    rpc_client
+        .send_and_confirm_transaction_with_spinner_and_config(
+            &tx,
+            config.commitment,
+            RpcSendTransactionConfig {
+                skip_preflight: true,
```
Comment (jstarry) [inline cli/src/program.rs:1249]: I see that other bpf loader transactions use `true` here but I don't see any reason for that. It'd be better to fail the transaction in preflight if the action isn't authorized or valid for some reason.

**Review by jstarry — COMMENTED** [review]
Looks great! Just a few nits

**Review by jstarry — APPROVED** [review]

**Comment by jstarry** [comment]
@0xripleys please merge master into your branch, the fix is in master already

**Comment by 0xripleys** [comment]
seemed like another flake to me so i just updated the branch. lmk if there's something else i should do

**Comment by jstarry** [comment]
Thanks, but most times there's no need to update the branch because most of the CI failures just require retrying an individual job. @yihau do you mind helping make sure this PR gets through CI? I don't have buildkite access at the moment

**Comment by yihau** [comment]
sure. will keep my eyes on this 👁️

**Pull request merged**